api/models/cartDao.js

```diff
@@ -0,0 +1,24 @@
+const dataSource = require("./dataSource");
+
+const createCart = async (userId, productId, quantity, price) => {
+  return await dataSource.query(
+    `INSERT INTO carts(
+      user_id,
+      product_id,
+      quantity,
+      price
+    ) VALUES (
+      ?,
+      ?,
+      ?,
+      ?
+    ) ON DUPLICATE KEY UPDATE
+      quantity = quantity + ${quantity}
+      `,
+    [userId, productId, quantity, price]
+  );
+};
```

Comment:
```suggestion
const createCart = async (userId, productId, quantity, price) => {
  return dataSource.query(
    `INSERT INTO carts(
      user_id,
      product_id,
      quantity,
      price
    ) VALUES (
      ?,
      ?,
      ?,
      ?
    ) ON DUPLICATE KEY UPDATE
      quantity = quantity + ?
      `,
    [userId, productId, quantity, price, quantity]
  );
};
```
현재 place holder 방법 (`?`를 활용하여 배열로서 변수의 위치를 지정하는것)과 템플릿 리터럴 (`${quantity}`) 방법을 혼용해서 같이 사용하고 있습니다. 하지만 sql injection이라는 해킹 공격과 연관된 보안상의 이유로 가급적이면 place holder를 사용하는 것을 추천하고 있습니다. 자세한 내용은 궁금하실 때 구두로 설명 드리겠습니다. 하나의 방법으로 통일시켜주세요~! :)